### ISSUE TITLE
Exception in pytorch classifier predict method

### ISSUE THREAD

**Issue by abigailgold** [Trusted-AI/adversarial-robustness-toolbox#1784]
**Describe the bug**
detach().cpu().numpy() is called twice in the following code (once before the if and once inside the if):

with torch.no_grad():
    model_outputs = self._model(torch.from_numpy(x_preprocessed[begin:end]).to(self._device))
output = model_outputs[-1]
output = output.detach().cpu().numpy().astype(np.float32)
if len(output.shape) == 1:
    output = np.expand_dims(output.detach().cpu().numpy(), axis=1).astype(np.float32)

This causes an exception. The internal call should be removed.

**System information (please complete the following information):**
 - OS: 12.4
 - Python version: 3.8
 - ART version or commit number: 1.11.0
 - TensorFlow / Keras / PyTorch / MXNet version
